Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,433 @@
+from . import Version
+from .Nodes import CNameDeclaratorNode
+from .ExprNodes import CallNode, NameNode, ImportNode, TupleNode, AttributeNode
+from ..CodeWriter import DeclarationWriter
+from .Visitor import CythonTransform
+from . import PyrexTypes
+from ..Utils import open_new_file
+import cython 
+import os 
+import sys 
+
+cython.declare(PyrexTypes=object, Naming=object, ExprNodes=object, Nodes=object,
+               Options=object, UtilNodes=object, LetNode=object,
+               LetRefNode=object, TreeFragment=object, EncodedString=object,
+               error=object, warning=object, copy=object, _unicode=object)
+
+
+# Inspired by and based around https://github.com/cython/cython/pull/3818
+# with some less lazy changes to it and a few minor improvements and optimizations...
+
+# Decided to revert to an older variant I had wrote of this code for the sake of 
+# maintainability - Vizonex
+
+
+
+# TODO Save this implementation commented out if required....
+if sys.version_info >= (3, 9):
```

Comment:
We don't assume that the version of Python that Cython is run in is necessarily the same as the version of Python the generated files are used in. Can this go in the stub files instead? (is this supported by stub files?)

Reply:
A try-import might work in the stub file.